src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +70,210 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
     configuration,
     toolkitOutputChannel,
     processInvoker = new DefaultSamCliProcessInvoker(),
     taskInvoker = new DefaultSamCliTaskInvoker()
-}: CodeLensProviderParams): void {
+}: CodeLensProviderParams): Promise<void> {
     const logger = getLogger()
 
-    const runtime = 'python3.7' // TODO: Remove hard coded value
+    const runtime = 'python3.6' // TODO: Remove hard coded value
 
-    const invokeLambda = async (args: LambdaLocalInvokeArguments) => {
+    const invokeLambda = async (args: LambdaLocalInvokeParams) => {
         const samProjectCodeRoot = await getSamProjectDirPathForFile(args.document.uri.fsPath)
-        logger.debug(`Project root: '${samProjectCodeRoot}'`)
+        const baseBuildDir = await makeBuildDir()
+
         let debugPort: number | undefined
 
+        let handlerName: string
+        let manifestPath: string | undefined
         if (args.isDebug) {
             debugPort = await getDebugPort()
+            const {debugHandlerName} = await makeLambdaDebugFile({
+                handlerName: args.handlerName,
+                debugPort: debugPort,
+                outputDir: samProjectCodeRoot,
+            })
+            handlerName = debugHandlerName
+            manifestPath = await makePythonDebugManifest({
+                samProjectCodeRoot,
+                outputDir: baseBuildDir
+            })
+        } else {
+            handlerName = args.handlerName
         }
-        // TODO: Figure out Python specific params and create PythonDebugConfiguration if needed
-        const debugConfig: DebugConfiguration = {
-            type: PYTHON_LANGUAGE,
-            request: 'attach',
-            name: 'SamLocalDebug',
-            preLaunchTask: undefined,
-            address: 'localhost',
-            port: debugPort!,
-            localRoot: samProjectCodeRoot,
-            remoteRoot: '/var/task',
-            skipFiles: []
-        }
+        const inputTemplatePath = await makeInputTemplate({
+            baseBuildDir,
+            codeDir: samProjectCodeRoot,
+            documentUri: args.document.uri,
+            handlerName,
+            runtime,
+            workspaceUri: args.workspaceFolder.uri
+        })
+        logger.info(`pythonCodeLensProvider.initialize: ${
+            JSON.stringify({samProjectCodeRoot, inputTemplatePath, handlerName, manifestPath}, undefined, 2)
+        }`)
+
+        const channelLogger = getChannelLogger(toolkitOutputChannel)
+        const codeDir = samProjectCodeRoot
+        const samTemplatePath: string = await executeSamBuild({
+            baseBuildDir,
+            channelLogger,
+            codeDir,
+            inputTemplatePath,
+            manifestPath,
+            samProcessInvoker: processInvoker,
 
-        const localLambdaRunner: LocalLambdaRunner = new LocalLambdaRunner(
+        })
+    
+        const debugConfig: PythonDebugConfiguration = makeDebugConfig({debugPort, samProjectCodeRoot})
+        await invokeLambdaFunction({
```

Comment:
this code feels familiar to `run` in localLambdaRunner. Is the run function hooked up anywhere?

Reply:
The functional `run` has been removed for now since was not used anywhere. If/when we decide to move the Nodejs CodeLens to use the functional approach and eliminate the `LocalLambdaRunner` class I (or someone else) can evaluate a re-usable functional `run`. I think it should be doable.